### PR TITLE
Enemies can now follow preset waypoints; Created spawner to spawn ene…

### PR DESCRIPTION
**Pull request by asori015**
…mies in hardcoded waves (for now)

### CODE REVIEW

File: Assets/Scripts/Spawner.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    private void Start(){
+        timer = timeBetweenWaves;
+        spawningWave = false;
+        waveIndex = 0;
+        enemyIndex = 0;
+        List<int> wave1 = new List<int>{0,0,0};
+        List<int> wave2 = new List<int>{0,0,0,0,0};
+        List<int> wave3 = new List<int>{0,0};
+        List<int> wave4 = new List<int>{0,0,0,0,0};
```
Comment (DishonJordan) [inline Assets/Scripts/Spawner.cs]: Do you think it would be better to make a wave object or something that allows it to be a little simple to add, edit, and change waves?

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "Build/":true,
+        "Library/":true,
+        "library/":true,
+        "obj/":true,
+        "Obj/":true,
+        "ProjectSettings/":true,
+        "temp/":true,
+        "Temp/":true
+    }
+}
```
Comment (DishonJordan) [inline .vscode/settings.json]: Is this file necessary?

File: Assets/Scenes/Main Scene.unity [hunk 3]
```diff
@@ -646,6 +686,54 @@ Transform:
...
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 204171319}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 926d6cd38ff19064eb75248b2238abdb, type: 3}
+  m_Name: 
+  m_EditorClassIdentifier: 
+  enemies:
+  - {fileID: 6083536873277876409, guid: b632eadc259dc4045a8b966bdf0542ae, type: 3}
+  startWaypoint: {fileID: 2130485517}
```
Comment (DishonJordan) [inline Assets/Scenes/Main Scene.unity]: When I started this up in Unity, the Waypoints gameobject in the spawner script on the Spawner object was empty. Double check

**Review by DishonJordan — CHANGES_REQUESTED** [review]
Nice work overall, just left a few comment about some improvements and one issue I ran into. Also do you think you could remove the enemy ufo that is in the scene by default?

File: Assets/Scripts/Enemy.cs [hunk 4]
```diff
@@ -12,8 +12,13 @@ public class Enemy : MonoBehaviour, IDamageable
     public float speed;
 
     [Tooltip("Current waypoint this enemy should move towards")]
-    public Transform targetWaypoint;
+    private Transform targetWaypoint;
+    private int waypointIndex;
+    public GameObject waypoints;
```
Comment (khuan028) [inline Assets/Scripts/Enemy.cs]: You can reference the Waypoint script directly instead of referencing the GameObject and using GetComponent.

File: Assets/Scripts/Spawner.cs (hunk 1 above)
Comment (asori015) [reply Assets/Scripts/Spawner.cs]: Will be done next sprint.
